### ISSUE TITLE
Dialogs: OS-Native vs. HTML-Native 

### ISSUE THREAD

**Issue by bajtos** [filecoin-project/filecoin-station#181]
We are using native OS dialog windows now. This creates several problems:

- Showing dialogs from the renderer (React frontend) via native OS API requires too much complexity in the IPC setup

- I think the UX would be nicer if we showed dialogs inside the HTML frontend, using the same visual design as the rest of the app. I feel the native OS dialog has a too different visual style.

- Can we show a different image from the default Electron icon? Here is a screenshot from macOS.

    <img width="340" alt="192775911-94b4dc00-6997-47dc-ba27-bc9413aab2f5" src="https://user-images.githubusercontent.com/1140553/193029275-444ee779-335c-4acf-aba3-1ebf9830c951.png">

**Comment by bajtos** [comment]
Question by @bajtos: 
> Can we show a different image from the default Electron icon? Here is a screenshot from macOS.
> <img width="340" alt="192775911-94b4dc00-6997-47dc-ba27-bc9413aab2f5" src="https://user-images.githubusercontent.com/1140553/193029275-444ee779-335c-4acf-aba3-1ebf9830c951.png">

Answer from @juliangruber:
> This is always the current app icon, so once it's bundled we see the Filecoin Station logo there :)

**Comment by juliangruber** [comment]
> * Showing dialogs from the renderer (React frontend) via native OS API requires too much complexity in the IPC setup

I don't think this should dictate the UX, if this is an actual problem then I'm sure we can create helpers that ease the process of adding another dialog.

> * I think the UX would be nicer if we showed dialogs inside the HTML frontend, using the same visual design as the rest of the app. I feel the native OS dialog has a too different visual style.

I'm really not sure about this one. Should we also reimplement the Save File As dialog? Also, looking for prior art, I saw that VSCode also uses native dialogs:

<img width="385" alt="Screenshot 2022-09-29 at 14 42 05" src="https://user-images.githubusercontent.com/10247/193034155-a714e631-366a-4d26-a319-6217730b2221.png">

**Comment by jopedroliveira** [comment]
we could implement a UI-based modal for the change wallet dialogue.
Triggering a native modal when clicking a button on the FE, instead of rendering an HTML component, doesn't feel natural to me.

Unless, ofc, we want to trigger that action from the tray or from some other part than the ui.

![Screenshot 2022-09-30 at 14 16 50](https://user-images.githubusercontent.com/6967377/193278038-a0ab21df-9987-4d94-8e37-c805be2c87e0.png)
